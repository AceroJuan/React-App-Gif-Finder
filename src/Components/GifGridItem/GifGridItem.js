import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn animate__delay-1s">
      <img src={url} alt={`info of ${title}`} />
      <p className="card__P">{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
