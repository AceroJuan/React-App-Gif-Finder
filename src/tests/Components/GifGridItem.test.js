import React from "react";
import "@testing-library/jest-dom";
import GifGridItem from "../../Components/GifGridItem/GifGridItem";
import { shallow } from "enzyme";

describe("pruebas en GifGridItem", () => {
  const title = "titulo de prueba";
  const url = "https://localhost:3000/image.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe mostrar <GifGridItem /> de manera correcta", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener la imagen igual al url y alt de props", () => {
    const img = wrapper.find("img");
    // console.log(img.props());
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe("info of " + title);
  });

  test("debe tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    // console.log(div.prop("className"));
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
