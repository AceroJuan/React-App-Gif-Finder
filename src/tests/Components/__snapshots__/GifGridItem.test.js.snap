// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe mostrar <GifGridItem /> de manera correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="info of titulo de prueba"
    src="https://localhost:3000/image.jpg"
  />
  <p
    className="card__P"
  >
    titulo de prueba
  </p>
</div>
`;
